docs/components/index.md

```diff
@@ -0,0 +1,337 @@
+
+Dynamic rendering logic (for example, loops, conditionals, expressions) are added using an embedded C# syntax called [Razor](https://docs.microsoft.com/aspnet/core/mvc/views/razor).
+
+When a Blazor app is compiled, the HTML markup and C# rendering logic are converted into a component class. The name of the generated .NET class matches the name of the file.
+
+Members of the component class are defined in a `@functions` block (more than one `@functions` block is permissible). In the `@functions` block, component state (properties, fields) is specified along with methods for event handling or for defining other component logic.
+
+Component members can then be used as part of the component's rendering logic using C# expressions that start with `@`. For example, a C# field is rendered by prefixing `@` to the field name. The following example evaluates and renders:
+
+* `_headingFontStyle` to the CSS property value for `font-style`.
+* `_headingText` to the content of the `<h1>` element.
+
+```cshtml
+<h1 style="font-style:@_headingFontStyle">@_headingText</h1>
+
+@functions {
+    private string _headingFontStyle = "italic";
+    private string _headingText = "Put on your new Blazor!";
+}
+```
+
+## Directives
+
+A component may specify one or more [directives](https://docs.microsoft.com/aspnet/core/mvc/views/razor#directives) at the top of the file.
+
+| Directive | Description |
+| --------- | ----------- |
+| [@functions](https://docs.microsoft.com/aspnet/core/mvc/views/razor#functions) | Adds a C# code block to a component. |
+| `@implements` | Implements an interface for the generated component class. Often used to create a component layout by implementing `ILayoutComponent`. |
+| [@inherits](https://docs.microsoft.com/aspnet/core/mvc/views/razor#inherits) | Provides full control of the class that the component inherits. |
+| [@inject](https://docs.microsoft.com/aspnet/core/mvc/views/razor#inject) | Enables service injection from the [service container](https://docs.microsoft.com/aspnet/core/fundamentals/dependency-injection). For more information, see [Dependency injection into views](https://docs.microsoft.com/aspnet/core/mvc/views/dependency-injection). |
+| `@layout` | Specifies a layout component. Layout components are used to avoid code duplication and inconsistency. |
+| [@page](https://docs.microsoft.com/aspnet/core/mvc/razor-pages#razor-pages) | Specifies that the component should handle requests directly. The `@page` directive can be specified with a route and optional parameters. Unlike Razor Pages, the `@page` directive doesn't need to be the first directive at the top of the file. |
+| [@using](https://docs.microsoft.com/aspnet/core/mvc/views/razor#using) | Adds the C# `using` directive to the generated component class. |
+| [@addTagHelper](xref:mvc/views/razor#tag-helpers) | Use `@addTagHelper` to use a component in a different assembly than the app's assembly. |
+
+## Databinding
```

Comment:
Before we get into data binding, I think we should have a section here called "Using components".

## Using components

Once you define a component it can be used to implement other components. The markup for using a component looks like an HTML tag where the name of the tag is the component type.

```html
<Counter />
```

Components can also have parameters, which you define using public properties on the component class. Use attributes to specify arguments for a component in markup.

```html
<Counter IncrementAmount="10" />
```

Reply:
We have the *Child components and component parameters* section. On the next commit, I'll take your language :point_up:, inject/revise that section with it, and move it above the *Data binding* section.